app/forms/openid_connect_token_form.rb

```diff
@@ -85,7 +85,7 @@ def validate_code_verifier
   def validate_client_assertion
     return unless identity.present?
 
-    service_provider = ServiceProvider.new(client_id)
+    service_provider = ServiceProvider.find_by_issuer!(client_id)
```

Comment:
I think this should be the non-! version of the method, or at least we might want to make something that has similar behavior to the old constructor that would make a like a null service provider